notifications/migrations/0009_alter_notification_data.py

```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+from django.db import models, migrations
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('notifications', '0008_index_together_recipient_unread'),
```

Comment:
if the library is being used and the version is upgraded this migration should migrate the data stored in the TextField to the JSONField in order to respect the integrity of the data, I recommend to do it in steps:

1. add a new JSONField field
2. Create an operation that passes the data from the TextField to the new JSONField.
3. Delete the TextField
4. Rename the field

Reply:
Yes, I can agree that direct ALTER FIELD is potentially dangerous in case of server fail during the ALTER process. And we can try to avoid db column to be messed up with duplicating it.
But, is it really worth it? Normally, you should dump your db data before potentially dangerous action - which is migration in our case.
With this migration we should warn users anyway about potential risks and recommend backup.

Reply:
I think a warning most of the time doesn't work hahaha people sometimes don't even specify the specific version and possibly an ‘automatic’ update could generate these problems.